libs/librepcbcommon/scopeguardlist.h

```diff
@@ -0,0 +1,66 @@
+/*
+ * LibrePCB - Professional EDA for everyone!
+ * Copyright (C) 2016 The LibrePCB developers
+ * http://librepcb.org/
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+
+
+#ifndef LIBREPCB_SCOPEGUARDLIST_H
+#define LIBREPCB_SCOPEGUARDLIST_H
+
+#include "scopeguard.h"
+#include <functional>
+#include <vector>
+
+namespace librepcb {
+
+class ScopeGuardList final : public ScopeGuardBase
+{
+    public:
+        ScopeGuardList() = default;
+
+        ScopeGuardList(ScopeGuardList&& rhs):
+            ScopeGuardBase(std::move(rhs)),
+            mScopeGuards(std::move(rhs.mScopeGuards))
+        { }
+
+        ScopeGuardList(const ScopeGuardList&) = delete;
+        ScopeGuardList& operator=(const ScopeGuardList&) = delete;
+
+        ~ScopeGuardList()
+        {
+            if (mActive) {
+                for (auto& scopeGuard : mScopeGuards) {
```

Comment:
I think the handlers should be called in reverse order, to match the behavior of multiple local `ScopeGuard` variables.